2_process.R

```diff
@@ -88,31 +88,48 @@ p2_targets_list <- list(
                   aoi_comids_df = p2_drb_comids_all_tribs)
   ),
   
   ## Estimate LC proportion in PRMS catchment
   # returns df with proportion LC in PRMS catchment in our AOI
   tar_target(p2_PRMS_NLCD_lc_proportions,
-             proportion_lc_by_prms(p2_NLCD_LC_w_catchment_area)),
+             proportion_lc_by_prms(p2_NLCD_LC_w_catchment_area) %>% select(-contains('NODATA'))),
 
   ## Standardize the land cover class names for NLCD to following standardized classes table - ''1_fetch/in/Reclassified_Land_Cover_IS.csv'
   # For NLCD, we use '1_fetch/in/Legend_NLCD_Land_Cover.csv' as vlookup file for the FORESCE targets
+
   tar_target(p2_PRMS_NLCD_lc_proportions_reclass,
-             reclassify_land_cover(land_cover_df = p2_PRMS_NLCD_lc_proportions,
-                                   reclassify_table_csv_path = '1_fetch/in/Legend_NLCD_Land_Cover.csv', 
-                                   reclassify_table_lc_col = 'NLCD_value', reclassify_table_reclass_col = 'Reclassify_match',
-                                   sep = ',',
-                                   pivot_longer_contains = 'NLCD') %>%
-               ## some lc classes in NLCD were given NA ultimately - example - alaska onluy shrub - we remove from table
-               select(-contains('NA')) %>% 
-               rename_with(~ gsub('prop_NLCD',"prop_lcClass", .x, fixed = T))
-  ),
-  
-  # Extract baccasted historical LC data raster values catchments polygond FORE-SCE  in the DRB - general function raster_to_catchment_polygons
+             ## spiting the df into list of dfs by individual years
+             purrr::map(.x = NLCD_year_suffix,
+                        .f = ~{p2_PRMS_NLCD_lc_proportions %>% select(PRMS_segid, AREASQKM_PRMS, contains(glue('NLCD', .x)))}
+                        ) %>% 
+              # reclassify by individual year df
+               purrr::map2(.x = .,
+                           .y = NLCD_year_suffix,
+                           .f = ~{reclassify_land_cover(land_cover_df = .x,
+                                                        reclassify_table_csv_path = '1_fetch/in/Legend_NLCD_Land_Cover.csv',
+                                                        reclassify_table_lc_col = 'NLCD_value',
+                                                        reclassify_table_reclass_col = 'Reclassify_match',
+                                                        sep = ',',
+                                                        pivot_longer_contains = glue('NLCD',.y)) %>% 
+                              # some lc classes in NLCD were given NA ultimately - example: alaska only shrub - we remove from table
+                              select(-contains('NA')) %>% 
+                              # Renaming col names 
+                               rename_with(.fn = function(x) sub("(_\\d+$)", "_lcClass\\1", x), .cols = starts_with("prop_"))
+                             }
+                           ) %>% 
+               # cbind all NLCD years
```

Comment:
Can you make the final dataset match the format of FORESCE? That had a column for each of the reclassified  LC classes and a column for the year of the dataset. Then all years were row bound.

Reply:
Putting this into a new Issue.

Reply:
Documenting what we discussed yesterday: reformatting to match the FORESCE table can be done in this PR

Reply:
Done in commit https://github.com/USGS-R/drb-inland-salinity-ml/pull/97/commits/08087e54d9ad38b43b52a9b3cbfe35d54d4899fb